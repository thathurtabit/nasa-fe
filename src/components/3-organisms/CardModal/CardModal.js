import React, { Component } from 'react';
import { connect } from 'react-redux';
import { TransitionGroup, CSSTransition } from 'react-transition-group';
import CardModalStyled, { CardModalBG } from './CardModal.styled';
import {
  CardContent,
  ShortDescription,
  CardLeft,
  CardRight,
} from '../Card/Card.styled';
import {
  loadDelay,
  NoDesc,
  NoTitle,
  BuyText,
} from '../../../utils/constants/constants';
import CloseModal from '../../1-atoms/CloseModal/CloseModal';
import LoadingSmall from '../../1-atoms/LoadingSmall/LoadingSmall';
import PageTitle from '../../1-atoms/PageTitle/PageTitle';
import CardImage from '../../1-atoms/CardImage/CardImage';
import Button from '../../1-atoms/Button/Button';

const mapStateToProps = state => ({
  products: state.response,
});

export class CardModal extends Component {
  constructor(props) {
    super(props);

    this.modalRef = React.createRef();

    this.state = {
      isLoading: true,
    };
  }

  componentDidMount() {
    const { location, products } = this.props;
    const productId = location.pathname
      .split('/')
      .filter(loc => loc)
      .pop();
    const productData = products.filter(
      product => product.productNo === productId
    )[0];

    this.setState({ product: productData, isLoading: false });

    this.modalRef.current.focus();
  }

  render() {
    const { product, isLoading } = this.state;

    return (
      <CardModalBG>
        <TransitionGroup>
          <CSSTransition classNames="modal-in" appear timeout={loadDelay}>
            <CardModalStyled
              data-card-modal
              role="dialog"
              aria-labelledby="modal-title"
              aria-modal="true"
              tabIndex="0"
              className="modal-in"
              ref={this.modalRef}
            >
              <CloseModal />
              {isLoading ? (
                <LoadingSmall isLoading />
              ) : (
                <CardContent>
                  <CardRight>
                    <PageTitle title={product.title || NoTitle} />
                    <ShortDescription>
                      {product.desc || NoDesc}
                    </ShortDescription>
                    <Button title={BuyText} url={product.link} external />
                  </CardRight>
                  <CardLeft>
                    <CardImage url={product.imgSrc} title={product.title} />
                  </CardLeft>
                </CardContent>
              )}
            </CardModalStyled>
          </CSSTransition>
        </TransitionGroup>
      </CardModalBG>
    );
  }
}

export default connect(mapStateToProps)(CardModal);

CardModal.defaultProps = {
  location: null,
};
