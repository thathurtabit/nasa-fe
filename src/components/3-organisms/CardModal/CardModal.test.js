import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { shallowWithTheme } from '../../../utils/test_config/testHelpers';

import {CardModal} from './CardModal';

describe('CardModal', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(
      <CardModal
        location='myLocation'
      />
    );
    expect(shallowToJson(element)).toMatchSnapshot();
  });
});
