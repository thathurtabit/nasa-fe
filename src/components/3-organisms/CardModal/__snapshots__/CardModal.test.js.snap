// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardModal should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
